packages/botonic-plugin-google-translate/src/index.ts

```diff
@@ -5,27 +5,37 @@ import { AccessToken } from './access-token'
 import { GoogleTranslateApiService } from './google-translate-api-service'
 import { LanguageDetector } from './language-detector'
 import { PluginOptions } from './options'
+import { Translator } from './translator'
 
 export default class BotonicPluginGoogleTranslate implements Plugin {
-  private readonly service: GoogleTranslateApiService
+  private readonly translator: Translator
   private readonly languageDetector: LanguageDetector
 
   constructor(readonly options: PluginOptions) {
-    this.service = new GoogleTranslateApiService(
+    const service = new GoogleTranslateApiService(
       new AccessToken(this.options.credentials),
       this.options.credentials.projectId
     )
+    this.translator = new Translator(service)
     this.languageDetector = new LanguageDetector(
-      this.service,
-      this.options.whitelist || []
+      service,
+      this.options.whitelist
     )
   }
 
   async pre(request: PluginPreRequest): Promise<void> {
     try {
       if (request.input.type == INPUT.TEXT && !request.input.payload) {
         const text = request.input.data
-        request.input.language = await this.languageDetector.detect(text)
+        if (this.options.translateTo) {
+          const translations = await this.translator.translate(
+            text,
+            this.options.translateTo
+          )
+          request.input['translations'] = translations
```

Comment:
Input type should have a translations field

Reply:
can the same strategy as before be applied here? import the basic input declaration and merge it with the fields that are appended here

Reply:
both comments refer to the same